applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -0,0 +1,419 @@
+        CheckProperty(DYNAMIC_VISCOSITY);
+        CheckProperty(BIOT_COEFFICIENT);
+        CheckProperty(PERMEABILITY_XX);
+    }
+
+    void CheckProperty(const Kratos::Variable<double>& rVariable) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable))
+            << rVariable.Name() << " does not exist in the thermal element's properties" << std::endl;
+        KRATOS_ERROR_IF(GetProperties()[rVariable] < 0.0)
+            << rVariable.Name() << " has an invalid value at element " << Id() << std::endl;
+    }
+
+    void CheckProperty(const Kratos::Variable<std::string>& rVariable, const std::string& rName) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable))
+            << rVariable.Name() << " does not exist in the thermal element's properties" << std::endl;
+        KRATOS_ERROR_IF_NOT(GetProperties()[rVariable] == rName)
+            << rVariable.Name() << " has a value of (" << GetProperties()[rVariable]
+            << ") instead of (" << rName << ") at element " << Id() << std::endl;
+    }
+
+    void CheckForNonZeroZCoordinateIn2D() const
+    {
+        if constexpr (TDim == 2) {
+            const auto& r_geometry = GetGeometry();
+            auto        pos        = std::find_if(r_geometry.begin(), r_geometry.end(),
+                                                  [](const auto& node) { return node.Z() != 0.0; });
+            KRATOS_ERROR_IF_NOT(pos == r_geometry.end())
+                << " Node with non-zero Z coordinate found. Id: " << pos->Id() << std::endl;
+        }
+    }
+
+    static void AddContributionsToLhsMatrix(MatrixType& rLeftHandSideMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rCompressibilityMatrix,
+                                            double DtPressureCoefficient)
+    {
+        rLeftHandSideMatrix = rPermeabilityMatrix;
+        rLeftHandSideMatrix += (DtPressureCoefficient * rCompressibilityMatrix);
+    }
+
+    void AddContributionsToRhsVector(VectorType& rRightHandSideVector,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rCompressibilityMatrix,
+                                     const array_1d<double, TNumNodes>& fluid_body_vector) const
+    {
+        const auto compressibility_vector =
+            array_1d<double, TNumNodes>{-prod(rCompressibilityMatrix, GetNodalValuesOf(DT_WATER_PRESSURE))};
+        rRightHandSideVector = compressibility_vector;
+        const auto permeability_vector =
+            array_1d<double, TNumNodes>{-prod(rPermeabilityMatrix, GetNodalValuesOf(WATER_PRESSURE))};
+        rRightHandSideVector += permeability_vector;
+        rRightHandSideVector += fluid_body_vector;
```

Comment:
Minor comment, but in my opinion it's a bit more clear if we make this a single line at the end:
rRightHandSideVector = compressibility_vector + permeability_vector + fluid_body_vector